src/Plugins/LevelDBStore/IO/Data/LevelDB/Native.cs

```diff
@@ -83,182 +91,169 @@ public static class Native
-        public static extern void leveldb_options_set_create_if_missing(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_create_if_missing(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_error_if_exists(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_error_if_exists(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_info_log(IntPtr /*Options*/ options, IntPtr /* Logger */ logger);
+        public static extern void leveldb_options_set_info_log(nint /*Options*/ options, nint /* Logger */ logger);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_paranoid_checks(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_paranoid_checks(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_env(IntPtr /*Options*/ options, IntPtr /*Env*/ env);
+        public static extern void leveldb_options_set_env(nint /*Options*/ options, nint /*Env*/ env);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_write_buffer_size(IntPtr /*Options*/ options, UIntPtr size);
+        public static extern void leveldb_options_set_write_buffer_size(nint /*Options*/ options, long size);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_max_open_files(IntPtr /*Options*/ options, int max);
+        public static extern void leveldb_options_set_max_open_files(nint /*Options*/ options, int max);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_cache(IntPtr /*Options*/ options, IntPtr /*Cache*/ cache);
+        public static extern void leveldb_options_set_cache(nint /*Options*/ options, nint /*Cache*/ cache);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_block_size(IntPtr /*Options*/ options, UIntPtr size);
+        public static extern void leveldb_options_set_block_size(nint /*Options*/ options, long size);
```

Comment:
Same here, These changes was tested in different architectures? @superboyiii

Reply:
I don't know if different machines were checked but I don't think it matters these are just sizes of the file on the machine

Reply:
it's a low level library, if the library expect an integer you can't send a long. we should check these changes (UIntPr/IntPtr to long/int)

Reply:
If it's going to be problematic. We can change it to it `unint`

Reply:
yes, `nuint` it's good